Use available Musl libc strtoull, strtoll, strtod, and strtof functions.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hemanth Kumar <hemanth@alifsemi.com>
--- bazel/output_base/external/flatbuffers/include/flatbuffers/util.h.orig	2025-11-30 15:24:47.817170646 +0530
+++ bazel/output_base/external/flatbuffers/include/flatbuffers/util.h	2025-11-30 15:24:26.070916275 +0530
@@ -225,10 +225,17 @@ inline std::string IntToStringHex(int i,
     #define __strtod_impl(s, pe) _strtod_l(s, pe, ClassicLocale::Get())
     #define __strtof_impl(s, pe) _strtof_l(s, pe, ClassicLocale::Get())
   #else
-    #define __strtoull_impl(s, pe, b) strtoull_l(s, pe, b, ClassicLocale::Get())
-    #define __strtoll_impl(s, pe, b) strtoll_l(s, pe, b, ClassicLocale::Get())
-    #define __strtod_impl(s, pe) strtod_l(s, pe, ClassicLocale::Get())
-    #define __strtof_impl(s, pe) strtof_l(s, pe, ClassicLocale::Get())
+    #ifdef __GLIBC__
+       #define __strtoull_impl(s, pe, b) strtoull_l(s, pe, b, ClassicLocale::Get())
+       #define __strtoll_impl(s, pe, b) strtoll_l(s, pe, b, ClassicLocale::Get())
+       #define __strtod_impl(s, pe) strtod_l(s, pe, ClassicLocale::Get())
+       #define __strtof_impl(s, pe) strtof_l(s, pe, ClassicLocale::Get())
+    #else
+       #define __strtoull_impl(s, pe, b) strtoull(s, pe, b)
+       #define __strtoll_impl(s, pe, b) strtoll(s, pe, b)
+       #define __strtod_impl(s, pe) strtod(s, pe)
+       #define __strtof_impl(s, pe) strtof(s, pe)
+    #endif
   #endif
 #else
   #define __strtod_impl(s, pe) strtod(s, pe)
